mod/InGameTracker/Logic.cs

```diff
@@ -393,6 +421,19 @@ public void DetermineAllAccessibility()
             AddConnection(TrackerRegions, reverseWarpConnection);
         }
 
+        // Conditionally add warp-based connection to Black Hole Forge
+        List<string> hourglassTwins = ["ET", "ST", "ETT", "ATT", "THT", "BHT", "GDT"];
+        List<string> brittleHollow = ["BHNG", "WHS"];
+        if (brittleHollow.Contains(bhfConnection) || (hourglassTwins.Contains(bhfConnection) && (hourglassTwins.Contains(bhngConnection) || hourglassTwins.Contains(whsConnection))))
```

Comment:
This condition is just hard enough to parse visually that I think it's worth breaking up into named local variables.

```suggestion
        // Conditionally add warp-based connection to Black Hole Forge
        List<string> hourglassTwins = ["ET", "ST", "ETT", "ATT", "THT", "BHT", "GDT"];
        List<string> brittleHollow = ["BHNG", "WHS"];

        bool hollowDirectlyConnectedToForge = brittleHollow.Contains(bhfConnection);
        bool hollowAndTwinsConnected = (hourglassTwins.Contains(bhngConnection) || hourglassTwins.Contains(whsConnection));
        bool hollowIndirectlyConnectedToForge = hourglassTwins.Contains(bhfConnection) && hollowAndTwinsConnected;
        if (hollowDirectlyConnectedToForge || hollowIndirectlyConnectedToForge)
```